maven_iuvs/echelle.py

```diff
@@ -841,3 +880,1530 @@ def find_files_with_geometry(file_index):
+    are for the standard resolution mode, not the echelle mode. 
+    
+    Parameters
+    ----------
+    lambda_c : float
+               a single wavelength in nm, candidate for a possible central wavelength marking the location of the emission peak.
+    bin_edges : array
+                Defines the bin edges in nm on which the LSF will be interpolated
+    CLSF : array
+           CLSF computed from the LSF in question, result of CLSF_from_LSF.
+
+    Returns
+    ----------
+    interp_CLSF : array
+                  the same CLSF reinterpolated on bin_edges.
+    """
+  
+    # Shift the wavelengths of the bin centers and edges
+    dlambda_binedges = binedges - lambda_c
+    
+    # Ensure that CLSF x is increasing everywhere so interp doesn't have meaningless results
+    if any(np.diff(CLSF) < 0):
+        raise Exception("ValueError: Can't interpolate because x values are not monotonically increasing")
+
+    interp_CLSF = np.interp(dlambda_binedges, CLSF[:, 0], CLSF[:, 1], left=0, right=1) 
+
+    # For some reason, interp function isn't automatically setting the edges to the requested values
+    if any(interp_CLSF > 1):
+        interp_CLSF[np.where(interp_CLSF > 1)] = 1
+
+    return interp_CLSF
+    
+
+def CLSF_from_LSF(LSFy, LSFx=None, dx=None):
+    """
+    Compute the empirical CLSF, given some LSF defined by xdata and ydata.
+    
+    Parameters
+    ----------
+    LSFy, LSFx : arrays
+                 x and y points which define the instrument LSF.
+    dx : real (float or int)
+         difference between consecutive x values. If not provided, will be calculated from xdata.
+
+    Returns
+    ----------
+    cumulative : array
+                 The CLSF, valid at the same xdata.
+    """
+
+    # The LSF usually comes in angstroms, and covers ± ~3 Å; put it in nm.
+    if 1 <= np.max(LSFx)<= 4:
+        LSFx = LSFx / 10
+
+    if dx is None:
+        if LSFx is None:
+            raise Exception("Please specify xdata or dx")
+        else:
+            dx = dx_array(LSFx) # Generate the dx array, since wavelengths are not equally spaced.
+        
+    cumulative = np.zeros((len(LSFx), 2))
+    
+    cumulative[:, 0] = LSFx
+    cumulative[:, 1] = np.cumsum(LSFy) * dx
+
+    # Normalize it so it asymptotes to 1.
+    cumulative[:, 1] = cumulative[:, 1] / cumulative[cumulative.shape[0]-1, 1]
+
+    return cumulative
+
+
+def load_lsf(calibration="new"):
+    """
+    Load appropriate LSF
+    """
+    lsf = sp.io.readsav(f"../IDL_pipeline/lsf_{calibration}.idl", idict=None, python_dict=False)
+    sav_var_names = {"new": ["echw", "echf"], 
+                     "old": ["w", "f"]
+                    }[calibration]
+    
+    lsfx_nm = lsf[sav_var_names[0]] / 10 # convert wavelength to nm, not angstrom
+    lsf_f = lsf[sav_var_names[1]]
+
+    return lsfx_nm, lsf_f
+
+
+def get_spectrum(data, light_fits, average=False, coadded=False, integration=0): 
```

Comment:
recommend adding a note that this applies only to the order containing H+D Lyman alpha

Reply:
Added in [17bd44f](https://github.com/lasp/maven_iuvs/pull/66/commits/17bd44fe74b3b51dce82c5966111ddc6d11f7dd0)